### PR TITLE
max.go

### PR DESCRIPTION
**Pull request by egor-geller**

### CODE REVIEW

File: max.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package main
+
+import "fmt"
+
+func max(x []string) string {
+	var l, s, p int
+	l = len(x)
+	s = len(x[0])
```
Comment (imatsiash-ib) [inline max.go]: Please use more obvious names for variables. I cannot understand from the variable name `s` what is stored in it.

Comment (imatsiash-ib) [reply max.go]: Are sure that slice `x` has an element with `0` index? 
We should check the length of slice.

File: max.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	for i := 1; i < l; i++ {
+		if s < len(x[i]) {
```
Comment (imatsiash-ib) [inline max.go]: Please use `utf8.RuneCountInString()` function to get the length of utf-8 string.

File: max.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	"unicode/utf8"
+)
+
+func max(x []string) string {
+	var stringLen, p int
+	if len(x) <= 0 {
+		fmt.Print("Slice has no elements")
+		os.Exit(1)
+	}
+	stringLen = len(x)
```
Comment (imatsiash-ib) [inline max.go]: This is a string length or a slice length?

File: max.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	maxCharLen := utf8.RuneCountInString(x[0])
+	for i := 1; i < stringLen; i++ {
+		if maxCharLen < len(x[i]) {
```
Comment (imatsiash-ib) [inline max.go]: Please use `utf8.RuneCountInString()` for x[i].

**Review by imatsiash-ib — APPROVED** [review]

File: max.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"fmt"
+	"os"
+	"unicode/utf8"
+)
+
+func max(x []string) string {
+	var sliceLen, p int
+	if len(x) <= 0 {
+		fmt.Print("Slice has no elements")
```
Comment (burov) [inline max.go:12]: It's better to return an error or panic

File: max.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+		os.Exit(1)
+	}
+	sliceLen = len(x)
```
Comment (burov) [inline max.go:15]: What happens if  slice an empty ?

File: max.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+	maxCharLen := utf8.RuneCountInString(x[0])
+	for i := 1; i < sliceLen; i++ {
+
+		if maxCharLen < utf8.RuneCountInString(x[i]) {
```
Comment (burov) [inline max.go:19]: Readability can be improved save result of utf8.RuneCountInString(x[I]) to variable and compare two variables

**Review by burov — APPROVED** [review]
It's ok for me, but implementation can be improved

**Pull request closed without merging**